OptimizelySDK/Optimizely.cs

```diff
@@ -308,5 +296,226 @@ public Variation GetForcedVariation(string experimentKey, string userId)
+        {
+            if (string.IsNullOrEmpty(featureKey))
+            {
+                Logger.Log(LogLevel.ERROR, "Feature flag key must not be empty.");
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(variableKey))
+            {
+                Logger.Log(LogLevel.ERROR, "Variable key must not be empty.");
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(userId))
+            {
+                Logger.Log(LogLevel.ERROR, "User ID must not be empty.");
+                return null;
+            }
+
+            var featureFlag = Config.GetFeatureFlagFromKey(featureKey);
+            if (string.IsNullOrEmpty(featureFlag.Key))
+                return null;
+
+            var featureVariable = featureFlag.GetFeatureVariableFromKey(variableKey);
+
+            if (featureVariable == null)
+            {
+                Logger.Log(LogLevel.ERROR,
+                    $@"No feature variable was found for key ""{variableKey}"" in feature flag ""{featureKey}"".");
+                return null;
+            }
+            else if (featureVariable.Type != variableType)
+            {
+                Logger.Log(LogLevel.ERROR,
+                    $@"Variable is of type ""{featureVariable.Type}"", but you requested it as type ""{variableType}"".");
+                return null;
+            }
+
+            var variableValue = featureVariable.DefaultValue;
+            var variation = DecisionService.GetVariationForFeature(featureFlag, userId, userAttributes);
+
+            if (variation != null && variation.GetFeatureVariableUsageFromId(featureVariable.Id) != null)
+            {
+                var featureVariableUsageInstance = variation.GetFeatureVariableUsageFromId(featureVariable.Id);
+                
+                variableValue = featureVariableUsageInstance.Value;
+                Logger.Log(LogLevel.INFO,
+                    $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureFlag.Key}"".");
+            }
+            else
+            {
+                Logger.Log(LogLevel.INFO,
+                    $@"User ""{userId}"" is not used in any variation for feature flag ""{featureFlag.Key}"", returning default value ""{variableValue}"".");
+            }
+
+            return variableValue;
+        }
+
+        /// <summary>
+        /// Gets boolean feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>bool | Feature variable value or null</returns>
+        public bool? GetFeatureVariableBoolean(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
```

Comment:
This will return either true or null. This is incorrect. 
We want to return false if TryParse does not parse to True

Reply:
TryParse will return true if value gets casted, false otherwise. This does not change the out parameter value.